### PR TITLE
Added tests and error pages

### PR DESCRIPTION
**Pull request by lodeme**
- some tests are incomplete due to undefined behavior. We need to discuss whether we want to implement them or not
- run tests using the following command:
python3 tests/run_all_tests.sh

### CODE REVIEW

File: src/requests/Request.cpp [hunk 1]
```diff
@@ -35,7 +36,8 @@ void Request::parse() {
         } else if (bytesRead == 0) {
             throw SocketCloseException("connection closed by client");
         } else {
-            throw ParsingErrorException(INTERRUPT, strerror(errno));
+			_badRequestFlag = 1;
+			return ;
```
Comment (ddavlet) [inline src/requests/Request.cpp:40]: Why don't we catch exception instead?

File: src/requests/Request.hpp [hunk 2]
```diff
@@ -37,13 +37,14 @@ class Request {
 	std::string getHeader(const std::string& key) const;
 	const std::map<std::string, std::string>& getHeaders() const;
 	std::string getBody() const;
+	bool getBadRequestFlag() const;
```
Comment (ddavlet) [inline src/requests/Request.hpp]: Do we need this line?

Comment (ddavlet) [reply src/requests/Request.hpp]: deleted

**Review by ddavlet — APPROVED** [review]
Looks good

**Review by olympiaobr — APPROVED** [review]

**Pull request merged**